docs/DeveloperGuide.md

```diff
@@ -19,6 +19,28 @@ Refer to the guide [_Setting up and getting started_](SettingUp.md).
 
 ### UI component
 
+![Structure of the UI Component](images/UiClassDiagram.png)
+
+**API**: `Ui.java`
+
+The UI consists of a `MainWindow` that is made up of parts e.g.`CommandBox`, `ResultDisplay`, `StatusBarFooter` etc. 
+All these, including the `MainWindow`, inherit from the abstract `UiPart` class.
+
+The `UI` component uses the JavaFx UI framework. The layout of these UI parts are defined in matching `.fxml` files 
+that are in the `src/main/resources/view` folder. For example, the layout of the `MainWindow` is specified in 
+`MainWindow.fxml`.
+
+The `UI` component:
+* Executes users' commands using the `Logic` component.
+* Listens for changes to `Model` data so that the UI can be updated with the modified data. 
+* Any changes in `Model` data, i.e. customers, cheeses or orders data, are reflected through the `Panels` and `Cards`
+  sub-components.
+  
+The class diagram below shows in more detail the compositions of the `Panels` and `Cards` components as well as their 
+relationships with other classes.
+
+![Structure of the Panels & Cards Component](images/UiPanelsCardsClassDiagram.png)
```

Comment:
Hi, may I know the link for the diagram

Reply:
I did it using plant UML leh (same as in AB3) so you will have to pull the branch to see it. I will paste them here so it's more convenient.

![UiClassDiagram](https://user-images.githubusercontent.com/77544764/113039855-b91c6800-91ca-11eb-9479-613ca38e1727.png)

![UiPanelsCardsClassDiagram](https://user-images.githubusercontent.com/77544764/113039875-be79b280-91ca-11eb-8f13-89f33cb52b39.png)

![ListPanelTogglingSequenceDiagram](https://user-images.githubusercontent.com/77544764/113039893-c46f9380-91ca-11eb-8b24-0d47f159bc21.png)

Reply:
I realized how ugly the panels and cards diagram is - gimme a min ill make some edits.

Reply:
![UiPanelsCardsClassDiagram](https://user-images.githubusercontent.com/77544764/113040381-4bbd0700-91cb-11eb-8287-ecb8429c1e29.png)

Reply:
This looks gd

Reply:
Should the sequence diagram contain the strings? it seems like a bit out of place